### ISSUE TITLE
Layout/SpaceAroundOperator causes false positives

### ISSUE THREAD

**Issue by tmtm** [rubocop/rubocop#13033]
Layout/SpaceAroundOperator causes false positives.

## Expected behavior

The following code should be passed:

```ruby
def foo
  xxx.each do
    a.foo.to_i +     # foo
      z.foo.to_i     # foo
  end
end
```

## Actual behavior

```
% rubocop foo.rb 
Inspecting 1 file
C

Offenses:

foo.rb:3:16: C: [Correctable] Layout/SpaceAroundOperators: Operator + should be surrounded by a single space.
    a.foo.to_i +     # foo
               ^

1 file inspected, 1 offense detected, 1 offense autocorrectable
```

but the following code passes even though the foo method is identical:

```ruby
def foo
  xxx.each do
    a.foo.to_i +     # foo
      z.foo.to_i     # foo
  end
end

def bar
  xxx.each do
    a.foo.to_i +
      z.foo.to_i
  end
end
```

## RuboCop version

```
% rubocop -V
1.64.1 (using Parser 3.3.3.0, rubocop-ast 1.31.3, running on ruby 3.3.3) [arm64-darwin22]
```
